### PR TITLE
Add Fluent implementation for Search service

### PR DESCRIPTION
**Pull request by milismsft**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by hovsepm** [comment]
@milismsft please pull from upstream and resolve conflicts.

File: src/ResourceManagement/Scheduler/Generated/JobsOperationsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+            /// <param name='jobCollectionName'>
+            /// The job collection name.
+            /// </param>
+            /// <param name='jobName'>
+            /// The job name.
+            /// </param>
+            public static JobDefinitionInner Get(this IJobsOperations operations, string resourceGroupName, string jobCollectionName, string jobName)
+            {
+                return operations.GetAsync(resourceGroupName, jobCollectionName, jobName).GetAwaiter().GetResult();
+            }
```
Comment (hovsepm) [inline src/ResourceManagement/Scheduler/Generated/JobsOperationsExtensions.cs]: please remove all Sync methods from the generated code. There is a regex expression that should help you.

File: src/ResourceManagement/Search/SearchServiceImpl.cs [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        {
+            return await this.Manager.Inner.Services.GetAsync(this.ResourceGroupName, this.Name, cancellationToken: cancellationToken);
+        }
+
+        ///GENMHASH:359B78C1848B4A526D723F29D8C8C558:4884876671AF0E98273703F5FEAB903C
+        protected async Task<Models.SearchServiceInner> CreateInnerAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (IsInCreateMode)
+            {
+            }
```
Comment (hovsepm) [inline src/ResourceManagement/Search/SearchServiceImpl.cs]: ? something missing in the if statement?

File: src/ResourceManagement/Search/SearchServiceImpl.cs [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+            return queryKeyList.Select(inner => new QueryKeyImpl(inner));
+        }
+
+
+        public async override Task<ISearchService> CreateResourceAsync(CancellationToken cancellationToken)
+        {
+            if (IsInCreateMode)
+            {
+            }
```
Comment (hovsepm) [inline src/ResourceManagement/Search/SearchServiceImpl.cs]: here as well. What is this check for?

**Comment by hovsepm** [comment]
Please add your new projects to https://github.com/Azure/azure-sdk-for-net/blob/Fluent/test_on_linux.sh#L34 file

**Comment by hovsepm** [comment]
Schedulers does not seems to have AssemblyInfo.cs file. Please add.

**Review by hovsepm — CHANGES_REQUESTED** [review]
please address review comments.

File: src/ResourceManagement/Search/AdminKeysImpl.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+namespace Microsoft.Azure.Management.Search.Fluent
```
Comment (brjohnstmsft) [inline src/ResourceManagement/Search/AdminKeysImpl.cs]: What is the relationship between this code and the generated code for our non-fluent .NET management SDK that lives in the `vs17dev` & `psSdkJson6` branches? If this is meant to be completely separate, why not have it in a separate repo?

Comment (milismsft) [inline src/ResourceManagement/Search/AdminKeysImpl.cs]: @martinsawicki
This class is the required implementation for the AdminKeys interface which is a wrapper for the *Inner respective class; in Fluent we do not want to expose the *Inner directly to the user.

**Review by hovsepm — APPROVED** [review]

**Comment by hovsepm** [comment]
sing-job: http://azuresdkci.cloudapp.net/view/NetCore-SDK/job/Fluent-NetCore-SDK-Sign/10/

File: src/ResourceManagement/Search/AdminKeysImpl.cs (hunk 4 above)
Comment (brjohnstmsft) [inline src/ResourceManagement/Search/AdminKeysImpl.cs]: @milismsft Sorry for the confusion; My question wasn't about this class in particular, it was about all these changes. I maintain the Search SDKs in the main branches (vs17dev & psSdkJson6) but I've been ignoring all things Fluent up until now. Is this code ever going to be merged back into the mainline branches, or is it permanently separate?

Comment (milismsft) [inline src/ResourceManagement/Search/AdminKeysImpl.cs]: @martinsawicki might be able to talk more about the future of Fluent branch but as of now the code stays in a separate branch out of the Azure/azure-sdk-for-net repo.

**Pull request merged**